Client/Frontend/Home/MessageCard/HomepageMessageCardViewModel.swift

```diff
@@ -94,18 +82,40 @@ extension HomepageMessageCardViewModel: HomepageViewModelProtocol {
     }
 
     var isEnabled: Bool {
+        return true
+    }
+
+    var hasData: Bool {
         return shouldDisplayMessageCard
     }
+
+    func updateData(completion: @escaping () -> Void) {
+        message = dataAdaptor.getMessageCardData()
+        completion()
+    }
 }
 
+// MARK: - HomepageSectionHandler
 extension HomepageMessageCardViewModel: HomepageSectionHandler {
 
     func configure(_ cell: UICollectionViewCell, at indexPath: IndexPath) -> UICollectionViewCell {
         guard let messageCell = cell as? HomepageMessageCardCell else {
             return UICollectionViewCell()
         }
 
+        handleMessageDisplayed()
```

Comment:
I remove the call to `handleMessageDisplayed` in the configure func because of the multiple calls to reload which trigger multiple impressions on the same app launch. There is a max amount of times we want to show the message but we should count only once per session